src/engine/debugging/DebugOverlay.cs

```diff
@@ -0,0 +1,243 @@
+﻿using System.Collections.Generic;
+using Godot;
+
+public class DebugOverlay : Control
+{
+    private readonly Dictionary<Microbe, Label> microbeLabels = new();
+    private readonly Dictionary<FloatingChunk, Label> floatingChunkLabels = new();
+
+    [Export]
+    private NodePath fpsCheckBoxPath = null!;
+
+    [Export]
+    private NodePath performanceMetricsCheckBoxPath = null!;
+
+    [Export]
+    private NodePath dialogPath = null!;
+
+    [Export]
+    private NodePath fpsCounterPath = null!;
+
+    [Export]
+    private NodePath performanceMetricsPath = null!;
+
+    [Export]
+    private NodePath labelsLayerPath = null!;
+
+    private bool showLabels;
+
+    private CustomDialog dialog = null!;
+    private CustomCheckBox fpsCheckBox = null!;
+    private CustomCheckBox performanceMetricsCheckBox = null!;
+    private FPSCounter fpsCounter = null!;
+    private PerformanceMetrics performanceMetrics = null!;
+    private Control labelsLayer = null!;
+    private Font smallerFont = null!;
+
+    private SceneTree rootTree = null!;
+    private Camera? activeCamera;
+
+    private bool ShowLabels
+    {
+        get => showLabels;
+        set
+        {
+            showLabels = value;
+            labelsLayer.Visible = value;
+        }
+    }
+
+    public override void _Ready()
+    {
+        fpsCheckBox = GetNode<CustomCheckBox>(fpsCheckBoxPath);
+        performanceMetricsCheckBox = GetNode<CustomCheckBox>(performanceMetricsCheckBoxPath);
+        dialog = GetNode<CustomDialog>(dialogPath);
+        fpsCounter = GetNode<FPSCounter>(fpsCounterPath);
+        performanceMetrics = GetNode<PerformanceMetrics>(performanceMetricsPath);
+        labelsLayer = GetNode<Control>(labelsLayerPath);
+        smallerFont = GD.Load<Font>("res://src/gui_common/fonts/Lato-Regular-Tiny.tres");
+
+        base._Ready();
+    }
+
+    public override void _EnterTree()
+    {
+        base._EnterTree();
+
+        Show();
+        InputManager.RegisterReceiver(this);
+        rootTree = GetTree();
```

Comment:
I think it's fine to just directly call `GetTree()` without caching it.

Reply:
Yeah, if it is called just once per frame, I would not see that as a problem.